pandas/io/parquet.py

```diff
@@ -260,8 +260,15 @@ def read_parquet(path, engine='auto', columns=None, **kwargs):
 
     Parameters
     ----------
-    path : string
-        File path
+    path : various
+        Either a path to a file (a :class:`str`,
```

Comment:
can you see what we say in other read_* methods and see if you can match them up

Reply:
I checked some: `read_hdf`, `read_json`, `read_csv`, `read_excel` and `read_feather`. They all seem different: 

* param  sometimes is `path`, in others `filepath_or_buffer` or `path_or_buf`
* param type is also different across methods. Probably `str, path object, URL or file-like object` would be the most complete and concise, but does it work for all methods?

Reply:
Those differences confuse the user quite a bit. I issued this PR when I found out I could actually use URLs also with `read_parquet`, and not just with `read_csv`.

Reply:
@jreback I could apply a custom variation of `str, path object, URL or file-like object` to each read_* method as a start. What about that?